cent/client/async_client.py

```diff
@@ -0,0 +1,115 @@
+from typing import List, Optional, Any, Dict, TypeVar
+
+from cent.client.session import BaseAsyncSession, AiohttpSession
+from cent.methods import (
+    CentMethod,
+    BroadcastMethod,
+    PublishMethod,
+    SubscribeMethod,
+)
+from cent.types import (
+    SubscribeObject,
+    StreamPosition,
+    PublishObject,
+    BroadcastObject,
+    Override,
+)
+
+T = TypeVar("T")
+
+
+class AsyncClient:
+    def __init__(
+        self,
+        base_url: str,
+        api_key: str,
+        session: Optional[BaseAsyncSession] = None,
+    ) -> None:
+        """
+        :param base_url: Centrifuge base_url
+        :param api_key: Centrifuge API key
+        :param session: Custom Session instance
+        """
+
+        self._base_url = base_url
+        self.api_key = api_key
+        self.session = session or AiohttpSession(base_url=base_url)
+
+    async def publish(
+        self,
+        channel: str,
+        data: Dict[str, Any],
+        skip_history: Optional[bool] = None,
+        tags: Optional[Dict[str, str]] = None,
+        b64data: Optional[str] = None,
+        idempotency_key: Optional[str] = None,
+        request_timeout: Optional[int] = None,
+    ) -> PublishObject:
+        call = PublishMethod(
+            channel=channel,
+            data=data,
+            skip_history=skip_history,
+            tags=tags,
+            b64data=b64data,
+            idempotency_key=idempotency_key,
+        )
+        return await self(call, request_timeout=request_timeout)
+
+    async def broadcast(
+        self,
+        channels: List[str],
+        data: Dict[str, Any],
+        skip_history: Optional[bool] = None,
+        tags: Optional[Dict[str, str]] = None,
+        b64data: Optional[str] = None,
+        idempotency_key: Optional[str] = None,
+        request_timeout: Optional[int] = None,
+    ) -> BroadcastObject:
+        call = BroadcastMethod(
+            channels=channels,
+            data=data,
+            skip_history=skip_history,
+            tags=tags,
+            b64data=b64data,
+            idempotency_key=idempotency_key,
+        )
+        return await self(call, request_timeout=request_timeout)
+
+    async def subscribe(
+        self,
+        user: str,
+        channel: str,
+        info: Optional[Dict[Any, Any]] = None,
```

Comment:
Same as data in publish - any JSON-encodable object is accepted, basically all `bytes` fields in https://github.com/centrifugal/centrifugo/blob/139e3085a95f7290be12a03e4310d07123bc5fb4/internal/apiproto/api.proto#L191 follow this. The rule is - when it's binary communication the field must be encoded as bytes, when it's JSON communication - the value is any embedded JSON. We have the same for client protocol BTW.